### ISSUE TITLE
Questions about the code

### ISSUE THREAD

**Issue by sasforce** [tomerraviv95/MetaViterbiNet#3]
1. The initial training (In 'meta_train' function) part has more than training 300 blocks.
2. Why is the pilot in initial training process not included in the buffer?
3. In the evaluation part (the 'eval_by_word' function),  SER is calculated using the labels for collecting meta-training data, but  it's less likely to know the labels of non-pilot. Then, is it wrong?

**Comment by tomerraviv95** [comment]
Thanks for reviewing the code!

1. For the paper's results I have used 20 minibatches, each batch contains 300 blocks. But see that in each minibatch the detector is trained on 10 values only (meta_j_num). As sample-efficiency was not my intent, only adapting from one distribution to another one I have not reduced the number of minibatches. However, you could try overfitting on a single or two minibatches and check the results (should be the same).
2. Setting the buffer_empty flag to false sets this (any minibatch from the train channel is what's important - not the same minibatch used for training, as highlighted in 1).
3. Due to the error correction code, you can use the errors_num calculated. See the original ViteribNet paper as well (algorithm 2).

Let me know if you have any more questions

**Comment by sasforce** [comment]
Thank you for your generous help!


Best regards


| |
ESCORT
|
|
***@***.***
|
签名由网易邮箱大师定制
On 9/25/2021 ***@***.***> wrote：

Thanks for reviewing the code!

For the paper's results I have used 20 minibatches, each batch contains 300 blocks. But see that in each minibatch the detector is trained on 10 values only (meta_j_num). As sample-efficiency was not my intent, only adapting from one distribution to another one I have not reduced the number of minibatches. However, you could try overfitting on a single or two minibatches and check the results (should be the same).
Setting the buffer_empty flag to false sets this (any minibatch from the train channel is what's important - not the same minibatch used for training, as highlighted in 1).
Due to the error correction code, you can use the errors_num calculated. See the original ViteribNet paper as well (algorithm 2).

Let me know if you any more questions

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
